Classes/Services/Document/DocumentManager.php

```diff
@@ -323,12 +329,34 @@ protected function updateRemotely($document, $workflowTransition = null, $delete
         }
 
         if ($this->getDocumentTransferManager()->update($document)) {
-            $this->removeDocument($document);
+
+            if(!$this->checkEmbargoDate($document)){
+                $this->removeDocument($document);
+            } else {
+                $document->setState(DocumentWorkflow::LOCAL_STATE_IN_PROGRESS . ':' . $document->getRemoteState());
```

Comment:
Having constants for document state is helpful. However constructing new state contants along the way by using string concatenation is going directly against this purpose. This must be encapsulated in a function.